contracts/tests/test_foreign_bridge.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+
+
+@pytest.fixture()
+def foreign_bridge_with_token_balance(
+    foreign_bridge_contract, tln_token_contract, premint_token_address
+):
+    transfer_amount = 1
+
+    tln_token_contract.functions.transfer(
+        foreign_bridge_contract.address, transfer_amount
+    ).transact({"from": premint_token_address})
+
+    assert (
+        tln_token_contract.functions.balanceOf(foreign_bridge_contract.address).call()
+        == transfer_amount
+    )
+
+    return foreign_bridge_contract
+
+
+def test_burn_token(foreign_bridge_with_token_balance, tln_token_contract):
+    assert (
+        tln_token_contract.functions.balanceOf(
+            foreign_bridge_with_token_balance.address
+        ).call()
+        > 0
+    )
+
+    foreign_bridge_with_token_balance.functions.burn().transact()
+
+    assert (
+        tln_token_contract.functions.balanceOf(
+            foreign_bridge_with_token_balance.address
+        ).call()
+        == 0
+    )
+
+
+def test_burn_token_with_zero_balance(foreign_bridge_with_token_balance):
```

Comment:
Shouldn't you check the balance here?

Reply:
I agree with you above comment, that we could avoid using the zero address.

Regarding the here mentioned balance check I decided willful against it, because this is not what should be tested. That the balance changes was tested in the case before. This makes sure the burn function can be called multiple times even if there is no balance left, without getting an assert or revert.
Would you still argue for it?

Reply:
I'm fine with leaving it as is, but don't you think checking for 0 would be nice to have?

Reply:
:heavy_check_mark: